app/models/users/permission_checks.rb

```diff
@@ -95,6 +95,26 @@ def all_permissions_for(context)
       .uniq
   end
 
+  # Helper method to be used in places where we just throw anything into the permission check and don't know what
+  # context it should be checked on. Things like menu item checks, controller action checks, generic services, etc.
+  def allowed_based_on_permission_context?(permission, project: nil, entity: nil) # rubocop:disable Metrics/PerceivedComplexity
```

Comment:
This method is not used in this PR but will be used to fix the few remaining instances